### PR TITLE
misc + iota axiom saves

### PR DESCRIPTION
**Pull request by icecream17**
Commit by commit (a lot more compartmentalized than my last pr)

About 40 to 50 miscallaneous downstream axiom saves are from the first few commits, and then iotaval/iotaex is responsible for the rest in the axiom diff.

### CODE REVIEW

File: set.mm
Comment (jkingdon) [inline set.mm]: Can we add a `$j usage` to provide a bit more specificity to "fewer axioms"?

File: set.mm
Comment (jkingdon) [inline set.mm:45190]: We want to keep the `$( $j usage 'dfopif' avoids 'ax-10' 'ax-11' 'ax-12'` even if we are restoring the old proof, right?

**Review by jkingdon — APPROVED** [review]
My suggestions are non-blocking.

The only thing in here which really caught my eye was the changes to iota related theorems. If someone who has thought more about iota than I has an opinion, speak up. If not, I think we can go with these changes, as they seem like improvements in terms of axiom usage.

File: set.mm
Comment (jkingdon) [inline set.mm:653934]: Can we add a $j usage to provide a bit more specificity to "fewer axioms"?

File: set.mm
Comment (jkingdon) [reply set.mm]: Something about the github mechanics led to this comment being called "outdated" so I made another comment. But both of them are about `ss2ab1` so you only need to pay attention to them once.

**Comment by icecream17** [comment]
~~blocked until #4503 is merged~~

File: set.mm
Comment (benjub) [inline set.mm]: Can you keep both versions ?  The old one is not completely subsumed and may be of interest.

**Review by benjub — APPROVED** [review]
Thanks @icecream17 for the careful updates !

**Pull request merged**
